File: 161278039/161278039/2/2.c
#include <stdio.h>

int main(){
	int n,a[100000],i,min,max;
	int q[3]={0},sum[3]={0},wait[3]={0};
	int qq[3]={0},j,ww[3]={0},t;
	scanf("%d",&n);
	for(i=0;i<n;i++)
		scanf("%d",&a[i]);
	
	for(i=0;i<n;i++){
		min=0;
		if(q[min]>q[1]) min=1;
		if(q[min]>q[2]) min=2;
		q[min]++;
		wait[min]+=sum[min];
		sum[min]+=a[i];
	}
	max=0;
	if(sum[max]<sum[1]) max=1;
	if(sum[max]<sum[2]) max=2;
	
	for(i=0;i<n;i++){
		min=0;
		if(qq[min]>qq[1]) min=1;
		if(qq[min]>qq[2]) min=2;
		ww[min]+=qq[min];
		qq[min]+=a[i];
	}
	j=0;
	if(ww[j]<ww[1]) j=1;
	if(ww[j]<ww[2]) j=2;
	t=0;
	if(qq[t]<qq[1]) t=1;
	if(qq[t]<qq[2]) t=2;
	printf("W1:%d T1:%d\n",wait[0]+wait[1]+wait[2],sum[max]);
	printf("W2:%d T2:%d\n",ww[j],qq[t]);
	return 0;
} 

File: 161220113/2.py
n=int(input("please input N:"))
strlist=input().split()
list1=[]
list2=[]
list3=[]
for i in strlist:
    i=int(i)
    temp=min(len(list1),len(list2),len(list3))
    if (temp==len(list1)):
        list1.append(i)
    elif (temp==len(list2)):
        list2.append(i)
    elif (temp==len(list3)):
        list3.append(i)
lenlist1=len(list1)-1
lenlist2=len(list2)-1
lenlist3=len(list3)-1
time1=0
time2=0
time3=0
for i in list1:
    time1+=i*lenlist1
    lenlist1=lenlist1-1
for i in list2:
    time2+=i*lenlist2
    lenlist2=lenlist2-1
for i in list3:
    time3+=i*lenlist3
    lenlist3=lenlist3-1
print("W1=",time1+time2+time3,"T1=",max(sum(list1),sum(list2),sum(list3)))
for i in strlist:
    i=int(i)
lis1=int(strlist[0])
lis2=int(strlist[1])
lis3=int(strlist[2])
i=3
time=0
while (i<=n-1):
    time=time+min(lis1,lis2,lis3)
    lis1=lis1-min(lis1,lis2,lis3)
    lis2=lis2-min(lis1,lis2,lis3)
    lis3=lis3-min(lis1,lis2,lis3)
    if(lis1==0):
        lis1=int(strlist[i])
        i=i+1
    if(lis2==0):
        lis2=int(strlist[i])
        i=i+1
    if(lis3==0):
        lis3=int(strlist[i])
        i=i+1
print("W2=",time,"T2=",time-int(strlist[n-1]))
        
    
    
    
    
        
    

